include/zephyr/data/json.h

```diff
@@ -591,6 +622,47 @@ int json_obj_parse(char *json, size_t len,
 	const struct json_obj_descr *descr, size_t descr_len,
 	void *val);
 
+/**
+ * @brief Parses the JSON-encoded object pointed to by @a json, with
+ * size @a len, and return the key-value pairs in the callback
+ * function @a raw_object.
+ *
+ * Since this parser is designed for machine-to-machine communications, some
+ * liberties were taken to simplify the design:
+ * (1) strings are not unescaped (but only valid escape sequences are
+ * accepted); and
+ * (2) no UTF-8 validation is performed;
+ *
+ * @param json Pointer to JSON-encoded value to be parsed
+ * @param len Length of JSON-encoded value
+ * @param raw_object Function to receive raw key value pairs from the JSON
+ *
+ * @return < 0 if error, 0 on success.
+ */
+int json_obj_parse_raw(char *json, size_t len,
```

Comment:
I would name the second parameter a bit differernt so it is clear that it is a callback

Reply:
Somehow my brain does not associatiate "object" with "callback".